crates/autopilot/src/domain/settlement/observation.rs

```diff
@@ -0,0 +1,36 @@
+//! Aggregated type, based on the mined settlement transaction.
+//!
+//! It contains all important information about the mined settlement, including
+//! the surplus and fees.
+
+// TODO
+
+use {
+    crate::domain::{auction, eth, settlement},
+    std::collections::HashMap,
+};
+
+pub struct Observation {
+    auction: auction::Id,
+    surplus: settlement::NormalizedSurplus,
+    fees: settlement::Fees,
+}
+
+impl Observation {
+    /// Creates a new observation from the given settlement and prices.
+    ///
+    /// No error handling since observation is sometimes partial but still
+    /// useful. ?
+    pub fn new(
+        settlement: &settlement::Encoded,
+        prices: &HashMap<eth::TokenAddress, auction::NormalizedPrice>,
```

Comment:
we should make it very clear that these are the pre-auction prices, maybe even pass in the whole `domain::auciton`? This will likely be needed to compute the fees anyways (for which we need fee policies, which are on the auction)

Reply:
What do you mean with pre-auction prices? (I saw already you mentioned that somewhere)? Is it before we divide it by 1e18?
auction::NormalizedPrice should represent already divided prices.

Reply:
moved normalization inside Trade as we did in the driver